### PR TITLE
refactor: split trade desk into audience and realtime conversions destinations

### PR DESCRIPTION
**Pull request by Gauravudia**
## What are the changes introduced in this PR?
- Split the Trade Desk into `audience` and `real time conversions` destinations. Where audience destination will be used for syncing audience from warehouse source through rETL flow and real time conversions destination will be used for tracking conversions through even stream flow.
- Created the separate folders `the_trade_desk` and `the_trade_desk_real_time_conversions`
- Moved the real time conversion on `processor`
- Transformer proxy is not required for real time conversions.

## What is the related Linear task?

Resolves INT-1558

## Please explain the objectives of your changes below

Put down any required details on the broader aspect of your changes. If there are any dependent changes, **mandatorily** mention them here

webapp and integration config PR's for separating out trade desk destination.
https://github.com/rudderlabs/rudder-webapp/pull/6057
https://github.com/rudderlabs/rudder-integrations-config/pull/1213
https://github.com/rudderlabs/rudder-integrations-config/pull/1212

### Any changes to existing capabilities/behaviour, mention the reason & what are the changes ?

N/A

### Any new dependencies introduced with this change?

N/A

### Any new generic utility introduced or modified. Please explain the changes.

N/A

### Any technical or performance related pointers to consider with the change?

N/A

<hr>

### Developer checklist

- [x] My code follows the style guidelines of this project

- [x] **No breaking changes are being introduced.**

- [x] All related docs linked with the PR?

- [x] All changes manually tested?

- [x] Any documentation changes needed with this change?

- [ ] Is the PR limited to 10 file changes?

- [x] Is the PR limited to one linear task?

- [x] Are relevant unit and component test-cases added?

### Reviewer checklist

- [x] Is the type of change in the PR title appropriate as per the changes?

- [x] Verified that there are no credentials or confidential data exposed with the changes.

## Summary by CodeRabbit

- **New Features**
	- Introduced integration with The Trade Desk real-time conversions API, including support for various e-commerce events and custom properties.
- **Refactor**
	- Streamlined utility functions focusing on essential tasks like TTL calculation, endpoint retrieval, and signature header generation.
	- Refactored proxy request handling to improve header management and response logic.
- **Bug Fixes**
	- Removed unsupported event type 'track' and related functionalities to enhance stability.
- **Chores**
	- Cleaned up test suites by removing outdated tests and adding new ones for real-time conversions integration.
- **Documentation**
	- Updated sample data and constants to align with The Trade Desk real-time conversions integration.

### CODE REVIEW

**Comment by Gauravudia** [comment]
@coderabbitai revie

**Comment by devops-github-rudderstack** [comment]
Test report for this run is available at: https://test-integrations-dev.s3.amazonaws.com/integrations-test-reports/rudder-transformer/3091/test-report.html

**Comment by devops-github-rudderstack** [comment]
Test report for this run is available at: https://test-integrations-dev.s3.amazonaws.com/integrations-test-reports/rudder-transformer/3091/test-report.html

**Comment by Gauravudia** [comment]
https://github.com/coderabbitai review

**Comment by Gauravudia** [comment]
@coderabbitai review

**Comment by koladilip** [comment]
Overall looks good, what was the main reason for this? is it came from product or more of tech decision?

**Comment by Gauravudia** [comment]
> Overall looks good, what was the main reason for this? is it came from product or more of tech decision?

Not feasible to support multiple dest types in a single destination due to `UI limitation`.

**Comment by devops-github-rudderstack** [comment]
Test report for this run is available at: https://test-integrations-dev.s3.amazonaws.com/integrations-test-reports/rudder-transformer/3091/test-report.html

**Comment by devops-github-rudderstack** [comment]
Test report for this run is available at: https://test-integrations-dev.s3.amazonaws.com/integrations-test-reports/rudder-transformer/3091/test-report.html

File: src/v0/destinations/the_trade_desk/networkHandler.js
[the remark this answers is not in the document]
Comment (Gauravudia) [reply src/v0/destinations/the_trade_desk/networkHandler.js:1]: No, batching is not supported and trade desk API return standard error codes

**Comment by krishna2020** [comment]
> Overall looks good, what was the main reason for this? is it came from product or more of tech decision?

We need some research on how to support multiple use-cases of a destination within one definition - this is not there as of today. And research needs UI layout changes along with backend handling. 
The way we have rETL flow of VDM screen showing different objects support, the same kind of behaviour need to be explored in Event Stream. This needs some time to figure out e2e flow. So for immediate resolution, going with separation now.

**Review by krishna2020 — APPROVED** [review]

**Review by koladilip — APPROVED** [review]

**Pull request merged**